File: src/main/java/org/social/social_network/entity/Post.java
package org.social.social_network.entity;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.MappedCollection;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;
import java.util.Set;
import java.util.UUID;

@Data
@Table(name = "posts")
public class Post {

    @Id
    private UUID id;

    private String description;

    private int likes;

    @CreatedDate
    private Instant pubDate = Instant.now();

    @LastModifiedDate
    private Instant modifiedDate = Instant.now();

    private UUID authorId;

    @MappedCollection(idColumn = "post_id")
    private Picture picture;

    @MappedCollection(idColumn = "post_id")
    private Set<Comment> comments;
}

File: src/main/java/org/social/social_network/entity/Post.kt
package org.social.social_network.entity

import lombok.Data
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.MappedCollection
import org.springframework.data.relational.core.mapping.Table
import java.time.Instant
import java.util.*

@Table(name = "posts")
data class Post(
    @Id
    var id: UUID? = null,
    var description: String? = null,
    var likes: Int = 0,
    @CreatedDate
    var pubDate: Instant = Instant.now(),
    @LastModifiedDate
    var modifiedDate: Instant = Instant.now(),
    var authorId: UUID? = null,
    @MappedCollection(idColumn = "post_id")
    var picture: Picture? = null,
    @MappedCollection(idColumn = "post_id")
    var comments: Set<Comment>? = null
)